src/data/Data.js

```diff
@@ -680,24 +692,27 @@ export const WorkspaceData = _.flow(
 
     const loadWdsData = useCallback(async () => {
       // Try to load the proxy URL
-      if (!wdsProxyUrl || !['Ready', 'Error'].includes(wdsProxyUrl.status)) {
-        const wdsUrl = await loadWdsUrl(workspaceId);
-        if (wdsUrl) {
-          await loadWdsTypes(wdsUrl, workspaceId);
+      if (!wdsApp || !['Ready', 'Error'].includes(wdsApp.status)) {
+        const wdsApp = await loadWdsApp(workspaceId);
+        // TODO: figure out how not to make this redundant fetch, per
+        // https://github.com/DataBiosphere/terra-ui/pull/4202#discussion_r1319145491
+        if (wdsApp) {
+          loadWdsTypes(wdsApp.proxyUrl);
```

Comment:
oopsie 1 again...proxyUrl isn't actually a field on app here, so this would always be undefined